deepchecks/utils/abstracts/weak_segment_abstract.py

```diff
@@ -96,8 +103,12 @@ def _create_heatmap_display(self, dummy_model, encoded_dataset, weak_segments, a
                         scores[f2_idx, f1_idx] = np.NaN
                         counts[f2_idx, f1_idx] = 0
                     else:
-                        scores[f2_idx, f1_idx] = scorer.run_on_data_and_label(dummy_model, segment_data,
-                                                                              segment_data[encoded_dataset.label_name])
+                        if scorer is not None and dummy_model is not None and label_col is not None:
+                            scores[f2_idx, f1_idx] = scorer.run_on_data_and_label(dummy_model,
```

Comment:
simplify further and make a equivalent function to run_on_data_and_label that accepts label and prediction rather than dummy model? Is there the issue with the indexes here that means we must have the dummy model?